pkg/services/secrets/kvstore/plugin.go

```diff
@@ -150,13 +175,15 @@ func (kv *secretsKVStorePlugin) GetAll(ctx context.Context) ([]Item, error) {
 	return parseItems(res.Items), err
 }
 
-func (kv *secretsKVStorePlugin) Fallback() SecretsKVStore {
-	return kv.fallback
+func (kv *SecretsKVStorePlugin) Fallback() SecretsKVStore {
+	return kv.fallbackStore
 }
 
-func (kv *secretsKVStorePlugin) SetFallback(store SecretsKVStore) error {
-	kv.fallback = store
-	return nil
+func (kv *SecretsKVStorePlugin) WithFallbackEnabled(fn func() error) error {
```

Comment:
Not necessary, but we could further eliminate the Fallback() function if the `WithFallbackEnabled` passed in func took a SecretsKVStore as an argument. Then the code calling this doesn't have to fetch it itself. If you decide to go that route, we should also guard against the fallback being nil

Reply:
We need to retrieve fallback to call the Del functions, and the delete process on fallback needs to happen after fallback has already been disabled and the secrets are being retrieved from the plugin store.

Reply:
ah okay, makes sense